src/main/java/online/book/store/service/BookServiceImpl.java

```diff
@@ -1,27 +1,38 @@
 package online.book.store.service;
 
 import java.util.List;
+import lombok.RequiredArgsConstructor;
+import online.book.store.dto.BookDto;
+import online.book.store.dto.CreateBookRequestDto;
+import online.book.store.exception.EntityNotFoundException;
+import online.book.store.mapper.BookMapper;
 import online.book.store.model.Book;
 import online.book.store.repository.BookRepository;
-import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.stereotype.Service;
 
 @Service
+@RequiredArgsConstructor
 public class BookServiceImpl implements BookService {
     private final BookRepository bookRepository;
+    private final BookMapper bookMapper;
 
-    @Autowired
-    public BookServiceImpl(BookRepository bookRepository) {
-        this.bookRepository = bookRepository;
+    @Override
+    public BookDto save(CreateBookRequestDto requestDto) {
+        Book book = bookMapper.toModel(requestDto);
+        return bookMapper.toDto(bookRepository.save(book));
     }
 
     @Override
-    public Book save(Book book) {
-        return bookRepository.save(book);
+    public List<BookDto> findAll() {
+        return bookRepository.findAll().stream()
+                .map(bookMapper::toDto)
+                .toList();
     }
 
     @Override
-    public List<Book> findAll() {
-        return bookRepository.findAll();
+    public BookDto findById(Long id) {
+        Book book = bookRepository.findById(id).orElseThrow(() ->
+                new EntityNotFoundException("Can't find book by id: " + id));
+        return bookMapper.toDto(book);
```

Comment:
```suggestion
         return bookMapper.toDto(bookRepository.findById(id).orElseThrow(()
                -> new EntityNotFoundException("Can't find book by id: " + id)));
```

Reply:
I thought that in this case it would be difficult to read, but when tried, looks good, thank you